Library/Dependencies/napi/source/env_chakra.cc

```diff
@@ -60,11 +60,9 @@ namespace Babylon
 
         ~napi_env_local()
         {
-            JsErrorCode errorCode = JsDisposeRuntime(m_jsRuntime);
-            if (errorCode != JsErrorCode::JsNoError)
-            {
-                Log(L"Failed to dispose runtime with error code (%u)\n", errorCode);
-            }
+            ThrowIfFailed(JsSetCurrentContext(JS_INVALID_REFERENCE));
```

Comment:
First and least controversial change, allows Chakra to shut down correctly. I have _not_ yet confirmed that this resolves the memory leak we were seeing before as there's a problem I just found with refresh; I'll update this comment when I fix that and confirm (or otherwise).

Reply:
Fixed the problems with refresh. Tested to confirm that refresh works, and has consistent and sane memory behavior, on Win32-Chakra, Win32-V8, and UWP-Chakra. UWP-V8 builds but does not run, which seems to be no change from master.

Reply:
#148